Mathlib/Order/Directed.lean

```diff
@@ -321,3 +321,65 @@ instance (priority := 100) OrderTop.to_isDirected_le [LE α] [OrderTop α] : IsD
 -- see Note [lower instance priority]
 instance (priority := 100) OrderBot.to_isDirected_ge [LE α] [OrderBot α] : IsDirected α (· ≥ ·) :=
   ⟨fun _ _ => ⟨⊥, bot_le _, bot_le _⟩⟩
+
+namespace DirectedOn
+
+section Pi
+
+namespace Pi
+
+variable {ι : Type*} {α : ι → Type*} [∀ i, LE (α i)]
+
+lemma proj {d : Set (Π i, α i)} (hd : DirectedOn (· ≤ ·) d) (i : ι) :
+    DirectedOn (· ≤ ·) ((fun a => a i) '' d) := by
+  intro p ⟨a, ha⟩ q ⟨b, hb⟩
+  obtain ⟨z,hz⟩ := hd a ha.1 b hb.1
+  use z i
+  aesop
```

Comment:
Could you instead add the following right after [Directed.mono_comp](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Directed.html#Directed.mono_comp) and use it here?
```lean
theorem DirectedOn.mono_comp {r : α → α → Prop} {rb : β → β → Prop} {g : α → β} {s : Set α}
    (hg : ∀ ⦃x y⦄, r x y → rb (g x) (g y)) (hf : DirectedOn r s) : DirectedOn rb (g '' s) :=
  directedOn_image.mpr (hf.mono hg)
```

Reply:
Also, I think this should simply be called `DirectedOn.proj` or `DirectedOn.eval`, double namespaces tend to be unusable.